File: backend/app/models/task.py
from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum


class TaskStatus(str, Enum):
    pending = "pending"
    in_progress = "in_progress"
    completed = "completed"
    cancelled = "cancelled"


class TaskPriority(str, Enum):
    low = "low"
    medium = "medium"
    high = "high"


@dataclass(slots=True)
class Task:
    id: int
    title: str
    description: str
    status: TaskStatus
    priority: TaskPriority
    owner: str
    tags: list[str] = field(default_factory=list)
    due_date: datetime | None = None
    created_at: datetime = field(default_factory=datetime.utcnow)
    updated_at: datetime = field(default_factory=datetime.utcnow)

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'in_progress' | 'completed' | 'cancelled'
export type TaskPriority = 'low' | 'medium' | 'high'

export interface Task {
  id: number
  title: string
  description: string
  status: TaskStatus
  priority: TaskPriority
  owner: string
  tags: string[]
  due_date: string | null
  created_at: string
  updated_at: string
}

export interface TaskPayload {
  title: string
  description: string
  status: TaskStatus
  priority: TaskPriority
  owner: string
  tags: string[]
  due_date: string | null
}

export interface TaskFilters {
  search: string
  status: TaskStatus | 'all'
}
